updater/aws.go

```diff
@@ -432,24 +433,36 @@ func (u *updater) sendCommand(instanceIDs []string, ssmDocument string) (string,
 	log.Printf("SSM document %q posted with command id %q", ssmDocument, commandID)
 
 	// Wait for the sent commands to complete.
-	errCount := 0
+	wg := sync.WaitGroup{}
+	instanceCount := len(instanceIDs)
+	errChan := make(chan error, instanceCount)
 	for _, v := range instanceIDs {
 		log.Printf("Waiting for command %q to complete for instance %q", commandID, v)
-		err = u.ssm.WaitUntilCommandExecutedWithContext(aws.BackgroundContext(), &ssm.GetCommandInvocationInput{
-			CommandId:  &commandID,
-			InstanceId: &v,
-		},
-			request.WithWaiterMaxAttempts(waiterMaxAttempts),
-			request.WithWaiterDelay(request.ConstantWaiterDelay(waiterDelay)))
-		if err != nil {
-			errCount++
-			log.Printf("Error encountered while awaiting document %q execution for instance: %q: %s", ssmDocument, v, err)
-			u.logCommmandOutput(commandID, v)
-		}
+		wg.Add(1)
+		go func(instanceID string) {
+			defer wg.Done()
+			err = u.ssm.WaitUntilCommandExecutedWithContext(aws.BackgroundContext(), &ssm.GetCommandInvocationInput{
+				CommandId:  aws.String(commandID),
+				InstanceId: aws.String(instanceID),
+			},
+				request.WithWaiterMaxAttempts(waiterMaxAttempts),
+				request.WithWaiterDelay(request.ConstantWaiterDelay(waiterDelay)))
```

Comment:
I'd prefer that we remove the constant delay here.  If there are concurrent calls and the calls end up being throttled, exponential backoff and a bit of jitter are both useful for smoothing the call rate.
```suggestion
```

Reply:
I verified that a max concurrency of 50 will not be an issue for the `GetCommandInvocation` API.

Reply:
If we remove `waiterDelay` overall wait time for SSM command can change because by default the [waiter Delay is 5 seconds](https://github.com/aws/aws-sdk-go/blob/v1.38.20/service/ssm/waiters.go#L32) and our `waiterDelay` is 15 seconds. Would "exponential backoff and a bit of jitter" help keep the same max wait time as before ?

Reply:
@srgothi92 Thanks for pointing that out; it looks like the waiter delay is constant by default.  My concern with a constant delay applies to both the default behavior and our 15-second change, so...  @WilboMo Can you put back the line I suggested you take out?  My apologies for the churn here...

(The concern with a constant delay and concurrent calls is that all the calls hit the SSM API in a very short window.  Since @WilboMo confirmed that 50 is not a problem, I'm comfortable having that be the case for now.  Longer term, I'd like to see something like a `JitteredWaiterDelay` to spread out the load a bit more.)

Reply:
Sure thing!